test/system/500-networking.bats

```diff
@@ -95,6 +95,12 @@ load helpers
     run_podman rm $cid
 }
 
+@test "podman run with slirp4ns assigns correct gateway address to host.gateway.internal" {
+    run_podman run --network slirp4netns:cidr=192.168.0.0/24,allow_host_loopback=true \
+                $IMAGE grep 192.168.0.2 /etc/hosts
+    is "$output"   "192.168.0.2 host.gateway.internal"   "host.gateway.internal should be the cidr+2 address"
+}
```

Comment:
I'm not feeling this behavior is correct.
The CIDR+2 address isn't useful when slirp4netns is launched with `--disable-host-loopback`.
And `--disable-host-loopback` should not be removed, as exposing bare 127.0.0.1 to containers might result in container-breakout.

I suggest exposing the "real" host IP regardless to rootful/rootless.

Reply:
So what would be the alternative? Are you suggesting to derive all interface ip addresses not including `lo`?

The issue (#5651) was to add behavior that reflects docker's `host.docker.internal`. 

I had a look through some of the docker code and as far as I can tell that is a configuration option given to the daemon so if a user asks for the host gateway address to be added to their container it is added based on that configuration [docker-ce reference](https://github.com/docker/docker-ce/blob/cc71e4fd9a9933078d79daaf69c8422c1bfa1618/components/engine/cmd/dockerd/config.go#L67)

So should podman have a similar configuration option?

Reply:
> Are you suggesting to derive all interface ip addresses not including lo?

The first interface (eth0) would be fine

Reply:
I'm not sure about this solution.

There can be so many different interfaces in so many different configurations that trying to derive an ip address from one of the interfaces seems unstable. I don't necessarily disagree that exposing the loopback interface to the container could result in a container breakout but this change doesn't give that access it simply puts a `/etc/hosts` entry for that interface.

Like I said above. I think the only viable alternative is to make the interface a configurable option similar to dockerd's `--host-gateway-ip` launch option.

Reply:
@AkihiroSuda bump in case you didn't see my last response

Reply:
Having an option SGTM